### PR TITLE
feat(KUI-1264): upgrade to KTH Style 10

### PR DESCRIPTION
**Pull request by karlandindrakryggen**
This PR upgrades KTH Style to version 10 and includes some more cleanup and fixes:

- Upgrade to KTH style packages 
  - Includes major cleanup of CSS
  - Adding same shared concept as in public web apps with copied files in `_shared.scss` and `components-shared` folder 
- Upgrade of other dependencies (minor/patch)
- Hardcoded toolbar URL are replaced with env variable
- Favicon path is fixed

https://kth-se.atlassian.net/browse/KUI-1264

### CODE REVIEW

**Comment by karlandindrakryggen** [comment]
Related PR in kurs-pm-admin-web: https://github.com/KTH/kurs-pm-admin-web/pull/276

**Review by axelbjo — APPROVED** [review]

**Review by amirhossein-haerian — APPROVED** [review]

File: test/messages.test.js [hunk 1]
```diff
@@ -47,11 +47,11 @@ const collectStuff = (object, doStuff, keysToIgnore = []) => {
  * @param key   Key name used for the current object
  * @param val   Current object to traverse in search of more keys
  */
-function doCollectKeys(data, key, val) {
+function doCollectKeys(data, key, val, keysToIgnore = []) {
   data.push(key)
   if (Array.isArray(val)) return
   if ('object' === typeof val) {
-    data.push(...collectKeys(val))
+    data.push(...collectKeys(val, keysToIgnore))
```
Comment (belanglos) [inline test/messages.test.js:54]: This recursion gave me a headache.
Assuming you have thought about this, are we sure that `keysToIgnore` is correct to pass on here? I.e. does it not only contain top-level keys?

**Review by belanglos — APPROVED** [review]
Good stuff! 🚀 
Looks like a lot of work 🐴, great work on extracting shared components!!

Some comments for understanding.

File: public/css/_shared.scss [hunk 2]
```diff
@@ -0,0 +1,166 @@
+// This file contains style shared (copied) between the admin apps of Om kursen apps (kurs-pm-admin-web, kurs-pm-data-admin-web, kursinfo-admin-web, kursutveckling-admin-web)
```
[the remark this answers is not in the document]
Comment (karlandindrakryggen) [reply public/css/_shared.scss:1]: I just checked, we did not include the link in the public apps, but I think we should! I will update here (and the other open PRs) now, and later we can add the link in corresponding file in public apps as well

File: public/js/app/components-shared/Alert.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/**
+ * https://intra.kth.se/style/en/components/alert
+ *
+ * Changelog:
+ * - 2024-04- - initial version (release of kth-style 10 public apps)
```
[the remark this answers is not in the document]
Comment (karlandindrakryggen) [reply public/js/app/components-shared/Alert.jsx:5]: Yeah, did have an exact day, so I just added the month for the initial files. I didn't have the Changelog comment in the public apps, but i think it's a good idea for the future if we starts to change these shared components before extracting then to a library.

File: server/views/course/index.handlebars [hunk 4]
```diff
@@ -4,8 +4,6 @@
 
 {{#if error}}
   <p>Error: {{error}}</p>
-{{else}}
```
[the remark this answers is not in the document]
Comment (karlandindrakryggen) [reply server/views/course/index.handlebars:7]: No, I couldn't find any usage of this api-data, so I removed it.

I didn't touch the invisible test-case.

File: test/messages.test.js (hunk 1 above)
Comment (karlandindrakryggen) [reply test/messages.test.js:54]: It gave me headache as well... but it felt like a bug to me at the time. But maybe I should have fixed this in another way? I think the keys I needed to ignore wasn't top-level.

File: public/js/app/components/AlertMsg.jsx [hunk 5]
```diff
@@ -49,8 +49,7 @@ const AlertMsg = ({ querySearchParams, courseCode, translate = {}, lang = 'en',
     : null
 
   return (
-    <Alert color="success" aria-live="polite">
```
[the remark this answers is not in the document]
Comment (karlandindrakryggen) [reply public/js/app/components/AlertMsg.jsx:52]: Fixed in Alert component as we talked about

**Pull request merged**